src/Node.js

```diff
@@ -0,0 +1,133 @@
+    self.parentNode = false;
+    self.nextSibling = false;
+    self.isVoid = false;
+    self.childNodes = [];
+    self.attributes = [];
+    Object.defineProperties(self, {
+      "innerHTML": {
+        get: function() {
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return null;
+          }
+
+          var innerHTML = "";
+
+          self.childNodes.forEach(function(childNode) {
+            innerHTML += childNode.outerHTML || "";
+          });
+
+          return innerHTML;
+        }
+      },
+      "outerHTML": {
+        get: function() {
+          if(self.nodeType === self.ATTRIBUTE_NODE) {
+            return null;
+          }
+          if(self.nodeType === self.COMMENT_NODE) {
+            return "<!--" + self.nodeValue + "-->";
+          }
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return self.nodeValue;
+          }
+
+          var attributeStr = "";
+          self.attributes.forEach(function(attr) {
+            attributeStr += attr.nodeName + "=\"" + (attr.nodeValue || "") + "\" ";
+          });
+          attributeStr = attributeStr.trim();
+          if(attributeStr.length > 0) {
+            attributeStr = " " + attributeStr;
+          }
+
+          var openingTag = "<" + self.nodeName.toLowerCase() + attributeStr + ">";
+          var closingTag = self.isVoid ? "" : "</" + self.nodeName.toLowerCase() + ">";
+
+          return openingTag + (self.innerHTML || "") + closingTag;
+        }
+      }
+    });
+  }
+
+  // Add a node to the current node as a child
+  Node.prototype.appendChild = function(node) {
+    var lastChild = this.childNodes.slice(-1)[0];
+    if(lastChild) {
+      lastChild.nextSibling = node;
+    }
+
+    node.parentNode = this;
+    this.childNodes.push(node);
+  };
+
+  Node.prototype.cloneNode = function() {
+    var node = new Node();
+    node.nodeName = this.nodeName;
+    node.nodeValue = this.nodeValue;
+    node.namespaceURI = this.namespaceURI;
+    node.nextSibling = this.nextSibling ? this.nextSibling.cloneNode() : false;
+    node.isVoid = this.isVoid;
+    this.childNodes.forEach(function(childNode) {
+      var clonedChildNode = childNode.cloneNode();
+      node.appendChild(clonedChildNode);
+    });
+    node.attributes = this.attributes.map(function(attribute) {
+      return attribute.cloneNode();
+    });
+    switch(this.nodeType) {
+      case this.ELEMENT_NODE:
+        node.nodeType = node.ELEMENT_NODE;
+        break;
+      case this.TEXT_NODE:
+        node.nodeType = node.TEXT_NODE;
+        break;
+      case this.COMMENT_NODE:
+        node.nodeType = node.COMMENT_NODE;
+        break;
+      case this.DOCUMENT_FRAGMENT_NODE:
+        node.nodeType = node.DOCUMENT_FRAGMENT_NODE;
+        break;
+    }
+
+    return node;
+  };
+
+  Node.prototype.getAttribute = function(name) {
+    if(this.nodeType !== this.ELEMENT_NODE) {
+      return;
+    }
+
+    var attrValue;
+
+    this.attributes.every(function(attr) {
```

Comment:
this code suggests you should not be using an array for attributes, but an object.